src/jest-extensions.js

```diff
@@ -75,6 +96,42 @@ const extensions = {
       }
     }
   },
+
+  toHaveAttribute(htmlElement, name, expectedValue) {
+    checkHtmlElement(htmlElement)
+    const isExpectedValuePresent = expectedValue === undefined
+    const hasAttribute = htmlElement.hasAttribute(name)
+    const receivedValue = htmlElement.getAttribute(name)
+    return {
+      pass: isExpectedValuePresent
+        ? hasAttribute && receivedValue === expectedValue
+        : hasAttribute,
+      message: () => {
+        const secondArgument = isExpectedValuePresent
+          ? printExpected(expectedValue)
+          : undefined
+        const to = this.isNot ? 'not to' : 'to'
+        const expectedAttribute = isExpectedValuePresent
+          ? expectedColor(`${name}=${stringify(expectedValue)}`)
+          : expectedColor(name)
+        const expectedMsg = `Expected the element ${to} have attribute ${expectedAttribute}`
+        const receivedMsg = getAttrReceivedMsg(
+          hasAttribute,
+          expectedValue,
+          receivedValue,
+        )
+        const matcher = matcherHint(
```

Comment:
I have used `assertMessage` method to handle the messages. Can't we re-use them? Like I have done it over here: https://github.com/gnapse/react-testing-library/blob/bfeb26aa29e2b36cbf254a6cb8dc6934e32a055e/src/jest-extensions.js#L89

Reply:
@antoaravinth Not in this case. The messages I wanted to generate do not comply with the predefined format that `assertMessage` uses.

BTW, not even the two already existing custom matchers used this function. It's only being used in one of them. I do not think we can expect all or even most matchers to always adhere to this expected/received constrained message format. I was checking at jest's own codebase and how they approach generating messages, and the most common functionality they have across all them is encoded in all the functions they also publish in `jest-matcher-utils` and that we're using here.

Reply:
Yes, I got it. Just wanted to check. Cool, thanks!